jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/index/GeometryLiteralIndex.java

```diff
@@ -60,17 +60,15 @@ private static GeometryWrapper retrieveMemoryIndex(String geometryLiteral, Geome
 
         if (INDEX_ACTIVE) {
 
-            if (index.containsKey(geometryLiteral)) {
-                geometryWrapper = index.get(geometryLiteral);
-            } else {
-                if (otherIndex.containsKey(geometryLiteral)) {
-                    geometryWrapper = otherIndex.get(geometryLiteral);
-                } else {
+            geometryWrapper = index.get(geometryLiteral);
+            if (geometryWrapper == null) {
+                geometryWrapper = otherIndex.get(geometryLiteral);
+                if (geometryWrapper == null) {
                     geometryWrapper = geometryDatatype.read(geometryLiteral);
                 }
                 index.put(geometryLiteral, geometryWrapper);
             }
```

Comment:
a matter of taste, but in Java 11 we could also write
``` java
geometryWrapper = index.putIfAbsent(geometryLiteral, otherIndex.getOrDefault(geometryLiteral, geometryDatatype.read(geometryLiteral)));
```

Reply:
This isn't quite the same - it always evaluates the inner function calls `otherIndex.getOrDefault` and `geometryDatatype.read`.

Reply:
True, my bad - should be `computeIfAbsent` to avoid unnecessary lookup of secondary index - so more like
``` java
geometryWrapper = index.computeIfAbsent(geometryLiteral, (lit) -> otherIndex.getOrDefault(lit, geometryDatatype.read(lit)));
```
would this be more readable? Don't know. Technically this is still more efficient from a Java internals point of view as it gathers the internal map bucket just once. But I don't think it matters in our case